hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecordPayload.java

```diff
@@ -58,6 +58,20 @@ default T preCombine(T oldValue, Properties properties) {
     return preCombine(oldValue);
   }
 
+  /**
+   * When more than one HoodieRecord have the same HoodieKey in the incoming batch, this function combines them before attempting to insert/upsert by taking in a schema.
+   * Implementation can leverage the schema to decide their business logic to do preCombine.
+   *
+   * @param oldValue   instance of the old {@link HoodieRecordPayload} to be combined with.
+   * @param schema Payload related schema. For example use schema to overwrite old instance for specified fields that doesn't equal to default value.
+   *
+   * @return the combined value
+   */
+  @PublicAPIMethod(maturity = ApiMaturityLevel.STABLE)
+  default T preCombine(T oldValue, Schema schema){
```

Comment:
Why can't we use `default T preCombine(T oldValue, Properties properties)` and pass schema from there? This API is not generic enough to be placed at this level.

Reply:
we can only pass string type in properties, if set schema String in props, need to parse it to Schema type when using it. I feel it would be a time-consuming operation since need to do it with every preCombine function. so I just add another function with the Schema parameter